rocketpy/Function.py

```diff
@@ -2024,9 +2024,267 @@ def integral(self, a, b, numerical=False):
+            value is the start of the dataset.
+        upper : scalar, optional
+            The upper limit of the interval in which the function is to be
+            plotted. If the Function is given by a dataset, the default
+            value is the end of the dataset.
+        datapoints : int, optional
+            The number of points in which the integral will be evaluated for
+            plotting it, which draws lines between each evaluated point.
+            The default value is 100.
+
+        Returns
+        -------
+        result : Function
+            The integral of the Function object.
+        """
+        # Check if lower and upper are given
+        if lower is None:
+            if isinstance(self.source, np.ndarray):
+                lower = self.source[0, 0]
+            else:
+                raise ValueError("Lower limit must be given if source is a function.")
+        if upper is None:
+            if isinstance(self.source, np.ndarray):
+                upper = self.source[-1, 0]
+            else:
+                raise ValueError("Upper limit must be given if source is a function.")
+
+        # Create a new Function object
+        xData = np.linspace(lower, upper, datapoints)
+        yData = np.zeros(datapoints)
+        for i in range(datapoints):
+            yData[i] = self.integral(lower, xData[i])
+        return Function(np.concatenate(([xData], [yData])).transpose(), 
+                    inputs=self.__inputs__, 
+                    outputs=[o + " Integral" for o in self.__outputs__])
+
+    def invert(self, lower=None, upper=None, datapoints=100):
+        """
+        Inverts the Function. Makes the domain the range and the range the domain.
+
+        Parameters
+        ----------
+        lower : float
+            Lower limit of the new domain.
+        upper : float
+            Upper limit of the new domain.
+
+        Returns
+        -------
+        result : Function
+            A Function whose domain and range have been inverted.
+        """
+        # Check if lower and upper are given
+        if lower is None:
+            if isinstance(self.source, np.ndarray):
+                lower = self.source[0, 0]
+            else:
+                raise ValueError("Lower limit must be given if source is a function.")
+        if upper is None:
+            if isinstance(self.source, np.ndarray):
+                upper = self.source[-1, 0]
+            else:
+                raise ValueError("Upper limit must be given if source is a function.")
+
+        # Create a new Function object
+        xData = np.linspace(lower, upper, datapoints)
+        yData = np.zeros(datapoints)
+        for i in range(datapoints):
+            yData[i] = self.getValue(xData[i])
+
+        try:
+            return Function(
+                np.concatenate(([yData], [xData])).transpose(),
+                inputs=self.__outputs__,
+                outputs=self.__inputs__,
+            )
```

Comment:
Is leaving interpolation and extrapolation arguments empty the best option we have here?